### ISSUE TITLE
IDE0052 fires on WithEvents that are used by Handles

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#30895]
``` VB
Private WithEvents _buildEvents As EnvDTE.BuildEvents
[...]

Private Sub BuildBegin(scope As EnvDTE.vsBuildScope, action As EnvDTE.vsBuildAction) Handles _buildEvents.OnBuildBegin

End Sub

```

Expected: No warning
Actual:

```
Severity	Code	Description	Project	Project Rank	File	Line	Category	Suppression State	Tool
Message	IDE0052	Private member 'ResourceEditorRootDesigner._buildEvents' can be removed as the value assigned to it is never read.	Microsoft.VisualStudio.Editors	3	E:\project-system2\src\Microsoft.VisualStudio.Editors\ResourceEditor\ResourceEditorRootDesigner.vb	70	Code Quality	Active	Compiler
```

**Comment by mavasani** [comment]
This seems like a core IOperation bug, analyzer doesn't seem to receive a callback for operations within the Handles clause.

**Comment by mavasani** [comment]
There are couple of underlying IOperation bugs causing the analyzer to function improperly:

1. Core compiler bug: https://github.com/dotnet/roslyn/issues/30978
2. Analyzer driver bug: https://github.com/dotnet/roslyn/issues/30979

Until these two bugs get fixed, I am going to put a simple workaround in this analyzer to handle VB handles clause specially. Once the above bugs are fixed, we can revert the analyzer workaround.

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/30980

**Comment by jbstcyr** [comment]
This is still an issue.

**Comment by davkean** [comment]
@jbstcyr This has been fixed in 16.0 Preview 2 which has not yet been released.
